bin/varnishd/mgt/mgt_jail_linux.c

```diff
@@ -48,11 +48,64 @@
 #include "mgt/mgt.h"
 #include "common/heritage.h"
 
+static int
+vjl_set_thp(const char *arg)
+{
+	int val;
+
+	if (!strcmp(arg, "ignore"))
+		return (0);
+	if (!strcmp(arg, "enable"))
+		val = 0;
+	else if (!strcmp(arg, "disable"))
+		val = 1;
+	else {
+		ARGV_ERR(
+		    "linux jail: unknown value '%s' for argument transparent_hugepage.\n",
+		    arg);
+	}
+	if (prctl(PR_SET_THP_DISABLE, val, 0, 0, 0) != 0) {
+		MGT_Complain(C_ERR,
+		    "Could not %s Transparent Hugepage: %s (%d)",
+		    arg, VAS_errtxt(errno), errno);
+	} else {
+		MGT_Complain(C_INFO, "linux jail: Transparent Hugepage now %sd",
+		    arg);
+	}
+	return (0);
+}
+
 static int
 vjl_init(char **args)
 {
+	char **unix_args;
+	int ret = 0;
+	size_t i;
+
+	if (args == NULL) {
+		/* Autoconfig */
+		if (vjl_set_thp("disable") != 0)
+			return (1);
+		return (jail_tech_unix.init(NULL));
+	}
 
-	return jail_tech_unix.init(args);
+	for (i = 0; args[i] != NULL; i++);
+	unix_args = calloc(i + 1, sizeof *unix_args);
+	AN(unix_args);
+
+	for (i = 0; *args != NULL && ret == 0; args++) {
+		if (!strncmp(*args, "transparent_hugepage=", 21)) {
```

Comment:
nit: can we please at least replace the magic 21 with something like
```c
	const char const *thpa = "transparent_hugepage=";
	const size_t l = strlen(thpa);
```

Reply:
FYI this was drawing inspiration from the UNIX jail.

Reply:
https://github.com/varnishcache/varnish-cache/commit/4001ea251bac6924c2e86bac3d0f2c8d93c96bd8
FTR, in the example above I got one `const` wrong.

Reply:
maybe `sizeof("transparent_hugepage=") - 1` would be appropriate and avoid calling `strlen` just for that ?

Reply:
Or `sizeof("transparent_hugepage")` to swallow the equal sign :p

A call to `strlen()` for a static string will likely be optimized out at compile time, but if anything I'd rather work with a macro here and in the UNIX jail:

```c
if ((val = ARG_MATCH(*args, transparent_hugepage)) != NULL)
        ret = vjl_set_thp(val);
```

Less ceremony, and the macro is in charge of dealing with the equal sign and value offset.

Reply:
I've added that as a separate commit and applied it to all jails, I hope its fine like that with you

Reply:
That was fast, but I only meant to make a suggestion here. I'm interested in @nigoroll's opinion on my suggestion, which would do more than the macro you pushed.

Reply:
@dridi 

> I'd rather work with a macro here and in the UNIX jail:

can you show how the macro would look like in `vju_init`? I thought about it but could not see a nice way cause of the "error if vju_get* fails, else continue"

Reply:
https://github.com/varnishcache/varnish-cache/pull/4202